### PR TITLE
Set group permission to shared subdirs

### LINKED ISSUE #268: permissions for sub-directories created in shared directory
Shared directory should be RW for all, which is how the permissions are set.
However, directories created under shared default to RO.  This can cause difficulty when trying to collaborate with others.

### PR DESCRIPTION
**Pull request by viniciusdc**
fix #268 

Whith the SetGID bit enabled, every  file created within the `shared` directory will  be automatically modified to be in the group of the directory (permission level).

### CODE REVIEW

**Comment by viniciusdc** [comment]
This should solve the problem with the groups permission on `shared`. It need a test though, I was not able to test it with the current status of Qhub #443

**Comment by costrouc** [comment]
Looks to be almost everything we need. I tested this locally and we also need to set https://stackoverflow.com/questions/55060753/how-do-i-set-a-default-umask-in-jupyterhub. I'm going to add to this PR tonight to add umasks.

**Comment by viniciusdc** [comment]
> Looks to be almost everything we need. I tested this locally and we also need to set https://stackoverflow.com/questions/55060753/how-do-i-set-a-default-umask-in-jupyterhub. I'm going to add to this PR tonight to add umasks.

Really, I tested it here and the permissions were there, have I missed something during that (thanks for the reply and test too)

File: qhub/template/{{ cookiecutter.repo_directory }}/templates/jupyterhub_config.py.j2 [hunk 1]
```diff
@@ -88,7 +88,7 @@ def qhub_configure_profile(username, safe_username, profile):
                   "chown {uid}:{primary_gid} /mnt/home/{username}",
                   "mkdir -p /mnt/home/shared",
                   "chmod 777 /mnt/home/shared"
```
Comment (viniciusdc) [inline qhub/template/{{ cookiecutter.repo_directory }}/templates/jupyterhub_config.py.j2:90]:
```suggestion
                  "chmod 2777 /mnt/home/shared"
```
Actually I think that I should have changed this line instead...

Comment (costrouc) [reply qhub/template/{{ cookiecutter.repo_directory }}/templates/jupyterhub_config.py.j2:90]: Nope you got the previous one completely right! I've been working on testing this PR.

Comment (viniciusdc) [reply qhub/template/{{ cookiecutter.repo_directory }}/templates/jupyterhub_config.py.j2:90]: yeah, indeed changing that line will not grant the correct groups permissions for each folder (instead I think root will receive it right?). Nvm, thanks for the reply

**Comment by costrouc** [comment]
So I had to do some extra work to get umasks to work with jupyterlab and update qhub in the process to get this working. See commits:
  - https://github.com/Quansight/qhub-cloud/commit/2ad4310d220de2664525cac9b1d77934b14ca8e9
  - https://github.com/Quansight/qhub-cloud/commit/69a3cd545f75e4bbaadae8442c2ef0c900ed4898

I tested on a digital ocean fresh install and got the following:

image

Going to close this PR. @viniciusdc your PR got this working and thanks for the discussion earlier about umask and setgid.

**Comment by viniciusdc** [comment]
> So I had to do some extra work to get umasks to work with jupyterlab and update qhub in the process to get this working. See commits:
> 
> * [2ad4310](https://github.com/Quansight/qhub-cloud/commit/2ad4310d220de2664525cac9b1d77934b14ca8e9)
> * [69a3cd5](https://github.com/Quansight/qhub-cloud/commit/69a3cd545f75e4bbaadae8442c2ef0c900ed4898)
> 
> I tested on a digital ocean fresh install and got the following:
> 
> image
> 
> Going to close this PR. @viniciusdc your PR got this working and thanks for the discussion earlier about umask and setgid.

Thanks for the feedback @costrouc !

**Pull request closed without merging**